packages/controller/src/WsServer.ts

```diff
@@ -239,6 +239,13 @@ ${err.stack}`
 	}
 
 	async registerHost(data: lib.RegisterHostData, socket: WebSocket, req: IncomingMessage) {
+		if (this.controller.recoveryMode) {
+			logger.warn(`Recovery | rejected host connection attempt for ${this.remoteAddr(req)}`);
+			wsRejectedConnectionsCounter.inc();
+			socket.close(lib.ConnectionClosed.TryAgainLater, "Connection Refused: Controller in recovery mode");
```

Comment:
Does TryAgainLater make sense if the host is in recovery mode? Trying again later won't produce a different result, unless the controller config is changed to not be in recovery mode and relaunched. It doesn't seem like a temporary thing to me.

I'd expect to see this response for rate limiting, intermittent network errors or the controller rejecting requests due to high load.

Reply:
I selected this code because recovery mode should be temporary (it is pretty useless putting the controller/host in recovery mode for an extended period). I like to think of it as a service becoming temporary unavailable for maintence.

This status was the closest fit without adding our own code; but there is no extra logic added on the client socket, so there is nothing preventing using our own code. So unless you have a suggestion for another then I will add `RecoveryMode = 4002` and use that.

Reply:
Recovery code makes a lot more sense to me